database/src/main/resources/db/migration/V1_163__fix_transport_service_search_result_view.sql

```diff
@@ -0,0 +1,39 @@
+-- Fix "external-interface-download-status" join to "external-interface-description"
+-- Drop view first
+DROP VIEW transport_service_search_result;
+
+-- Create view again
+CREATE VIEW transport_service_search_result AS
+SELECT t.*, op.name as "operator-name", op."business-id" as "business-id",
+       -- Changed Start
+       array_cat(
+               array_cat(
+                       (SELECT sc."companies"
+                        FROM "service_company" sc
+                        WHERE sc."transport-service-id" = t.id),
+                       t."companies"),
+               array_agg((aso.name, aso."business-id")::company)) AS "service-companies",
+       -- Changed End
+       (SELECT array_agg(oaf."operation-area")
+        FROM "operation-area-facet" oaf
+        WHERE oaf."transport-service-id" = t.id) AS "operation-area-description",
+       (SELECT array_agg(ROW(ei."external-interface", ei.format,
+           ei."ckan-resource-id",
+           ei.license, ei."data-content",
+           eids."download-error",
+           eids."db-error")::external_interface_search_result)
+
+               -- Changed section
+        FROM "external-interface-description" ei
+                 LEFT JOIN (SELECT DISTINCT ON (eids.id) eids.id, eids."external-interface-description-id",
```

Comment:
As an afterthought, this SELECT DISTINCT needs some additional love :)
eids.id doesn't make sense here.